src/components/sbb-timetable-occupancy-icon/sbb-timetable-occupancy-icon.ts

```diff
@@ -0,0 +1,89 @@
+import { CSSResult, html, LitElement, PropertyValues, TemplateResult } from 'lit';
+import { customElement, property, state } from 'lit/decorators.js';
+import style from './sbb-timetable-occupancy-icon.scss?lit&inline';
+import { OccupancyEnum } from '../core/timetable';
+import { documentLanguage, HandlerRepository, languageChangeHandlerAspect } from '../core/eventing';
+import { i18nOccupancy } from '../core/i18n';
+import '../sbb-icon';
+
+@customElement('sbb-timetable-occupancy-icon')
+export class SbbTimetableOccupancyIcon extends LitElement {
+  public static override styles: CSSResult = style;
+
+  /** Wagon occupancy. */
+  @property() public occupancy!: OccupancyEnum;
+
+  /** Negative coloring variant flag. */
+  @property({ type: Boolean }) public negative: boolean = false;
+
+  /** The icon name which will be rendered. */
+  @state() private _iconName: string;
+
+  @state() private _currentLanguage = documentLanguage();
+
+  private _handlerRepository = new HandlerRepository(
+    this,
+    languageChangeHandlerAspect((l) => (this._currentLanguage = l)),
+  );
+
+  private _setIconName(): void {
+    if (!this.occupancy) {
+      return;
+    }
+
+    if (window.matchMedia('(forced-colors: active)').matches) {
+      // high contrast
+      this._iconName = `utilization-${this._transformOccupancy()}-high-contrast`;
+    } else if (
+      this.hasAttribute('negative') ||
+      window.matchMedia('(prefer-color-scheme: dark)').matches
+    ) {
+      // dark
+      this._iconName = `utilization-${this._transformOccupancy()}-negative`;
+    } else {
+      this._iconName = `utilization-${this._transformOccupancy()}`;
+    }
+  }
+
+  private _transformOccupancy(): string {
+    return !this.occupancy || this.occupancy.toLowerCase() === 'unknown'
+      ? 'none'
+      : this.occupancy.toLowerCase();
+  }
+
+  public override connectedCallback(): void {
+    super.connectedCallback();
+    this._setIconName();
+    window.matchMedia('(forced-colors: active)').onchange = () => this._setIconName();
+    window.matchMedia('(prefer-color-scheme: dark)').onchange = () => this._setIconName();
+    this._handlerRepository.connect();
+  }
+
+  protected override willUpdate(changedProperties: PropertyValues<this>): void {
+    if (changedProperties.has('occupancy') || changedProperties.has('negative')) {
+      this._setIconName();
+    }
+  }
+
+  public override disconnectedCallback(): void {
+    super.disconnectedCallback();
+    this._handlerRepository.connect();
+  }
+
+  protected override render(): TemplateResult {
+    return html`
+      <sbb-icon name=${this._iconName}></sbb-icon>
+      <span class="sbb-timetable-occupancy-icon--visually-hidden">
+        ${i18nOccupancy[this.occupancy?.toLowerCase()] &&
+        `${i18nOccupancy[this.occupancy.toLowerCase()][this._currentLanguage]}`}
+      </span>
```

Comment:
Could this not just placed as aria-label on the sbb-icon? (and setting aria-hidden=false)

Reply:
With the current state, it's impossible for a customer to somehow overwrite aria-labels or set another role as the icon is completely wrapped. If it would be inherited, it could receive the same behavior as the icon itself, so it's easier for the customer.

Reply:
With commit f1f71c2adae2aeef8e60842932b57a22e3de1511 the sbb-occupancy-icon extends the sbb-icon, which should solve the problem